cgen/preprocessor.py

```diff
@@ -0,0 +1,70 @@
+from cgen.infrastructure import Generable, Line, Module
+
+
```

Comment:
Could you include a `__doc__ = ...` here, to work with the `automodule`?

Reply:
you mean adding the infrastructure module in the ```doc/source/cgen.rst```?